Improved multi-field accessibility and usability.

jonearley opened the pull request:
This PR is a result from a conversation started over in Sufia: https://github.com/projecthydra/sufia/issues/2021#issuecomment-219096513.

mjgiarlo commented:
To confirm: @jonearley has a signed CLA, so his work is clear to merge.

jcoyne commented on app/assets/javascripts/hydra-editor/manage_repeating_fields.js:
```diff
@@ -3,139 +3,151 @@
+          this.options.label = ' ' + label;
+        }
+
+        this.controls = $("<span class=\"input-group-btn field-controls\">");
```
This should be passed in from the options.  Did you start from an out of date commit?

jcoyne commented:
This PR appears to revert the commit: https://github.com/projecthydra/hydra-editor/commit/be2ad3b7cdbba4fcb23ee78e15ab4b7f2ae0d471

I was rather fond of that commit, so can you work it back in?

jcoyne commented:
Here's a few other things to take a look at:
1. Please take a look at the Jasmine tests, they appear to need some adjustment.
1. Please update your commit to explain what you did to improve accessibility.
1. Please add tests for the features you have added.

jonearley commented:
@jcoyne I've added back in the be2ad3b commit (mostly). Since the label is added into the add and remove buttons I've swapped out the html option to a text option as a compromise.

jcoyne commented:
@jonearley still think you should make the html a template that goes into the options.  It could be function that takes the text as an argument, or you could use something like [Handlebars](http://handlebarsjs.com/)

jonearley commented:
@jcoyne I've updated the PR so the add and remove buttons can be passed in as a html option. I added a span to select on and insert text. It works, but maybe there is a better way.

jcoyne commented on app/assets/javascripts/hydra-editor/manage_repeating_fields.js:
```diff
@@ -3,139 +3,171 @@
+        $addHtml.find('.controls-add-text').html(options.addText + this.options.label);
+        this.adder    = $addHtml;
+
+        var $removeHtml = $(options.removeHtml);
```
please encapsulate this (and the following 2 lines) into a function.

jcoyne commented on app/assets/javascripts/hydra-editor/manage_repeating_fields.js:
```diff
@@ -3,139 +3,171 @@
+        this.options = options;
+
+        this.options.label = '';
+        var $label = this.element.find("label");
```
Please move lines 11-16 into a function.

jonearley replied:
Good idea. My recent force push reflects your suggestion.

jcoyne commented on app/assets/javascripts/hydra-editor/manage_repeating_fields.js:
```diff
@@ -3,139 +3,171 @@
+        },
+
+        _newField: function ($activeField) {
+            var $newField = this.createNewField($activeField);
```
Is there any reason to assign to a variable, or can you just return the result of the function call?

jonearley replied:
Good catch, I pushed a change to fix this.
